{
  "errors": [],
  "generated_at": "2024-10-27T18:06:08Z",
  "metrics": {
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\setup.py": {
      "CONFIDENCE.HIGH": 3,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 2,
      "SEVERITY.MEDIUM": 1,
      "SEVERITY.UNDEFINED": 0,
      "loc": 99,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tensorflow_metadata\\__init__.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 1,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tensorflow_metadata\\proto\\__init__.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 8,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tensorflow_metadata\\proto\\v0\\__init__.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 1,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tensorflow_metadata\\python\\proto_test.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 13,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tensorflow_metadata\\version.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 2,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tools\\build_tfmd_docs.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 1,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 1,
      "SEVERITY.UNDEFINED": 0,
      "loc": 40,
      "nosec": 0,
      "skipped_tests": 0
    },
    "_totals": {
      "CONFIDENCE.HIGH": 3,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 1,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 2,
      "SEVERITY.MEDIUM": 2,
      "SEVERITY.UNDEFINED": 0,
      "loc": 164,
      "nosec": 0,
      "skipped_tests": 0
    }
  },
  "results": [
    {
      "code": "18 import shutil\n19 import subprocess\n20 \n",
      "col_offset": 0,
      "end_col_offset": 17,
      "filename": "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\setup.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Consider possible security implications associated with the subprocess module.",
      "line_number": 19,
      "line_range": [
        19
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/blacklists/blacklist_imports.html#b404-import-subprocess",
      "test_id": "B404",
      "test_name": "blacklist"
    },
    {
      "code": "75   def run(self):\n76     subprocess.check_call(\n77         [self._bazel_cmd, 'run',\n78          '--compilation_mode', 'opt',\n79          *self._additional_build_options,\n80          '//tensorflow_metadata:move_generated_files'],\n81         # Bazel should be invoked in a directory containing bazel WORKSPACE\n82         # file, which is the root directory.\n83         cwd=os.path.dirname(os.path.realpath(__file__)),)\n84 \n",
      "col_offset": 4,
      "end_col_offset": 57,
      "filename": "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\setup.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "LOW",
      "issue_text": "subprocess call - check for execution of untrusted input.",
      "line_number": 76,
      "line_range": [
        76,
        77,
        78,
        79,
        80,
        81,
        82,
        83
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b603_subprocess_without_shell_equals_true.html",
      "test_id": "B603",
      "test_name": "subprocess_without_shell_equals_true"
    },
    {
      "code": "87   globals_dict = {}\n88   exec(fp.read(), globals_dict)  # pylint: disable=exec-used\n89 \n",
      "col_offset": 2,
      "end_col_offset": 31,
      "filename": "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\setup.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "MEDIUM",
      "issue_text": "Use of exec detected.",
      "line_number": 88,
      "line_range": [
        88
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b102_exec_used.html",
      "test_id": "B102",
      "test_name": "exec_used"
    },
    {
      "code": "35 \n36 _OUTPUT_DIR = flags.DEFINE_string('output_dir', '/tmp/tfmd_api/',\n37                                   'The path to output the files to')\n38 \n",
      "col_offset": 48,
      "end_col_offset": 64,
      "filename": "D:\\bandit\\upstream-repos\\py3-tensorflow-metadata\\tools\\build_tfmd_docs.py",
      "issue_confidence": "MEDIUM",
      "issue_cwe": {
        "id": 377,
        "link": "https://cwe.mitre.org/data/definitions/377.html"
      },
      "issue_severity": "MEDIUM",
      "issue_text": "Probable insecure usage of temp file/directory.",
      "line_number": 36,
      "line_range": [
        36,
        37
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b108_hardcoded_tmp_directory.html",
      "test_id": "B108",
      "test_name": "hardcoded_tmp_directory"
    }
  ]
}